logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/mention/MessageMentionMapper.kt

```diff
@@ -41,38 +41,50 @@ class MessageMentionMapperImpl(
         return MessageMention(
             start = mention.start,
             length = mention.length,
-            userId = mention.userId.toModel(),
-            isSelfMention = mention.userId.toModel() == selfUserId
+            userId = mention.userId?.toModel(),
+            isSelfMention = mention.userId?.toModel() == selfUserId
         )
     }
 
     override fun fromModelToDao(mention: MessageMention): MessageEntity.Mention {
         return MessageEntity.Mention(
             start = mention.start,
             length = mention.length,
-            userId = mention.userId.toDao()
+            userId = mention.userId?.toDao()
         )
     }
 
-    override fun fromProtoToModel(mention: Mention): MessageMention {
-        val userId = mention.qualifiedUserId?.let {
-            idMapper.fromProtoUserId(it)
-        } ?: UserId(
-            mention.mentionType?.value as String,
-            selfUserId.domain
-        )
-        return MessageMention(
+    override fun fromProtoToModel(mention: Mention): MessageMention = mention.qualifiedUserId?.let {
```

Comment:
How about making this return an optional `MessageMention` instead and do a `mapNotNull` when unpacking a text message.